From e99c9c97266d6d3e8cac798aa91408250c4d60cf Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 29 Mar 2013 09:00:34 +0400
Subject: [PATCH 17/35] gcc-poison-dir-extend

Add /sw/include and /opt/include based on the original
zecke-no-host-includes.patch patch.  The original patch checked for
/usr/include, /sw/include and /opt/include and then triggered a failure and
aborted.

Instead, we add the two missing items to the current scan.  If the user
wants this to be a failure, they can add "-Werror=poison-system-directories".

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending
---
 gcc/incpath.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gcc/incpath.c b/gcc/incpath.c
index cd41c78..eac4a92 100644
--- a/gcc/incpath.c
+++ b/gcc/incpath.c
@@ -393,7 +393,9 @@ merge_include_chains (const char *sysroot, cpp_reader *pfile, int verbose)
          {
           if ((!strncmp (p->name, "/usr/include", 12))
               || (!strncmp (p->name, "/usr/local/include", 18))
-              || (!strncmp (p->name, "/usr/X11R6/include", 18)))
+              || (!strncmp (p->name, "/usr/X11R6/include", 18))
+              || (!strncmp (p->name, "/sw/include", 11))
+              || (!strncmp (p->name, "/opt/include", 12)))
             warning (OPT_Wpoison_system_directories,
                      "include location \"%s\" is unsafe for "
                      "cross-compilation",
-- 
1.7.10.4

